src/ingestion/sources/bigquery.py

```diff
@@ -0,0 +1,200 @@
+import json
+
+from ingestion.task import TaskUnit
+
+from .base import MetricsQueryBuilder, SourceConfiguration, SQLAlchemySourceDialect, SQLAlchemySource
+from sqlalchemy.engine.url import make_url
+
+class BigQueryMetricsQueryBuilder(MetricsQueryBuilder):
+    def _base_query_sample(self, select_sql):
+        return """
+            SELECT 
+                CURRENT_TIMESTAMP as `WINDOW_START`, 
+                CURRENT_TIMESTAMP as `WINDOW_END`, 
+                COUNT(*) as `ROW_COUNT`, 
+                '{table}' as `TABLE_NAME`,
+                '{database}' as `DATABASE_NAME`,
+                '{schema}' as `SCHEMA_NAME`,
+
+                {select_sql}
+            FROM {schema}.{table};
+        """.format(
+            select_sql=select_sql,
+            table=self.monitor['table_name'],
+            schema=self.monitor['schema'],
+            database=self.monitor['database'],
+        )
+
+    def _base_query_backfill(self, select_sql, table, timestamp_field):
+        return """
+            SELECT 
+                DATE_TRUNC({timestamp_field}, HOUR) as `WINDOW_START`, 
+                TIMESTAMP_ADD(DATE_TRUNC({timestamp_field}, HOUR), INTERVAL 1 HOUR) as `WINDOW_END`, 
+                COUNT(*) as `ROW_COUNT`, 
+                {table} as `TABLE_NAME`,
+                {database} as `DATABASE_NAME`,
+                {schema} as `SCHEMA_NAME`,
+                {timestamp_field} as `TIMESTAMP_FIELD`,
+
+                {select_sql}
+            FROM {schema}.{table} as c
+            WHERE 
+                DATE_TRUNC({timestamp_field}, HOUR) >= TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL {minutes_ago} MINUTE)
```

Comment:
CURRENT_TIMESTAMP() here vs. CURRENT_TIMESTAMP above

Did we check that backfill query executes correctly? If not, we should just not include it for now

Reply:
Yes the query was checked and executes successfully. `CURRENT_TIMESTAMP` and `CURRENT_TIMESTAMP()` both pull the timestamp in BQ, but we will use `CURRENT_TIMESTAMP()` per the docs recommendations - https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions

Reply:
should include database here as well